### PR TITLE
Feature/dsnd 465 initialise spring boot application for charges processor

### PR DESCRIPTION
**Pull request by gseregni-ch**
Initialise Spring Boot Application for Charges Processor - dsnd-465
Creation of BDD Framework (using Cucumber) Charges Processor/Consumer - dsnd-500

### CODE REVIEW

**Comment by gseregni-ch** [comment]
DSND-465 - Initialise spring boot application for charges processor. This PR resolves also DSND-500 - Creation of BDD Framework (using Cucumber) Charges Processor/Consumer.

File: src/itest/java/uk/gov/companieshouse/charges/delta/consumer/ChargesDeltaConsumerITest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @Autowired
+    public KafkaTemplate<String, ChsDelta> kafkaTemplate;
+
+    @Value("${charges.delta.topic.main}")
+    private String mainTopic;
+
+    @Test
+    public void testSendingKafkaMessage() {
+        ChsDelta chsDelta = new ChsDelta("{ \"key\": \"value\" }", 1, "some_id");
+        kafkaTemplate.send(mainTopic, chsDelta);
```
Comment (vchitti-kainos) [inline src/itest/java/uk/gov/companieshouse/charges/delta/consumer/ChargesDeltaConsumerITest.java:21]: This test does not test or verify anything, can you please remove. Or verify by consuming it.

File: src/main/java/uk/gov/companieshouse/charges/delta/producer/ChargesDeltaProducer.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import uk.gov.companieshouse.kafka.producer.Acks;
+import uk.gov.companieshouse.kafka.producer.CHKafkaProducer;
+import uk.gov.companieshouse.kafka.producer.ProducerConfig;
+
+@Component
+public class ChargesDeltaProducer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ChargesDeltaProducer.class);
+
+    @Value("${spring.kafka.bootstrap-servers}")
+    private String bootstrapServers;
```
Comment (vchitti-kainos) [inline src/main/java/uk/gov/companieshouse/charges/delta/producer/ChargesDeltaProducer.java:20]: inject using constructor for easy testable.

File: src/test/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    //    void When_ValidChsDeltaMessage_Expect_ValidChargesDeltaMapping() throws IOException {
+    //        Message<ChsDelta> mockChsDeltaMessage = createChsDeltaMessage();
+    //        //TODO To change the InsolvencyDelta class to ChargesDelta when this will be available
+    //        InsolvencyDelta expectedChargesDelta = createChargesDelta();
+    //        when(transformer.transform(expectedChargesDelta)).thenCallRealMethod();
+    //
+    //        deltaProcessor.processDelta(mockChsDeltaMessage);
+    //
+    //        verify(transformer).transform(expectedChargesDelta);
+    //    }
```
Comment (vchitti-kainos) [inline src/test/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessorTest.java:51]: Remove commented code above.

File: src/test/java/uk/gov/companieshouse/charges/delta/serialization/ChsDeltaDeserializerTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Test
+    void When_deserialize_Expect_ValidChsDeltaObject() {
+        ChsDelta chsDelta = new ChsDelta("{\"key\": \"value\"}", 1, "context_id");
+        byte[] data = encodedData(chsDelta);
+
+        ChsDelta deserializedObject = deserializer.deserialize("", data);
+
+        assertThat(deserializedObject).isEqualTo(chsDelta);
+    }
```
Comment (vchitti-kainos) [inline src/test/java/uk/gov/companieshouse/charges/delta/serialization/ChsDeltaDeserializerTest.java:23]: Can you please include a test if deserialiation fails.

File: src/itest/java/uk/gov/companieshouse/charges/delta/consumer/ChargesDeltaConsumerITest.java (hunk 1 above)
Comment (gseregni-ch) [reply src/itest/java/uk/gov/companieshouse/charges/delta/consumer/ChargesDeltaConsumerITest.java:21]: This is just a sample test. We will remove/enhance when picking up functional stories.

File: src/main/java/uk/gov/companieshouse/charges/delta/producer/ChargesDeltaProducer.java (hunk 2 above)
Comment (gseregni-ch) [reply src/main/java/uk/gov/companieshouse/charges/delta/producer/ChargesDeltaProducer.java:20]: It's a good point but as it's already merged we will make the changes in the next PR.

File: src/test/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessorTest.java (hunk 3 above)
Comment (gseregni-ch) [reply src/test/java/uk/gov/companieshouse/charges/delta/processor/ChargesDeltaProcessorTest.java:51]: It's a good point but as it's already merged we will remove this in the next PR.

File: src/test/java/uk/gov/companieshouse/charges/delta/serialization/ChsDeltaDeserializerTest.java (hunk 4 above)
Comment (gseregni-ch) [reply src/test/java/uk/gov/companieshouse/charges/delta/serialization/ChsDeltaDeserializerTest.java:23]: This is just a sample unit test and the related test will be implemented separately in the functional story.

**Pull request merged**